pkg/git/git_test.go

```diff
@@ -0,0 +1,204 @@
+//go:build integration
+// +build integration
+
+package git
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+)
+
+// All test cases in this file are integration tests that rely on a remote git
+// repository. They're (for now) disabled by default.
+//
+// Run these against a repository you don't mind being messed with.
+//
+// To use your own repository and credentials, set env vars:
+// - TEST_REPO_URL
+// - TEST_REPO_USERNAME
+// - TEST_REPO_PASSWORD (personal access token)
```

Comment:
For future enhancements / a bit of advice: the way I have tested these kind of things before is using https://github.com/sosedoff/gitkit.

Reply:
Oh... that is slick. I'm going to give that a try in the morning because I would much prefer that these tests didn't require a real repo.

Reply:
Fixed in 445089d901a785dd4a000d0956d6252284a2b4c4

Still working on other feedback.